src/net.cpp

```diff
@@ -2278,6 +2315,14 @@ bool CConnman::Start(CScheduler& scheduler, const Options& connOptions)
         }
     }
 
+    m_anchors_db_path = GetDataDir() / "anchors.dat";
+    m_anchors = ReadAnchors(m_anchors_db_path);
```

Comment:
> It seems we need to re-write the anchors.dat file every time when another anchor is dropped in m_anchors.pop_back();, no?

I think it would be simpler to delete the contents of the whole file once you have loaded the anchors into memory. Eg. After you read anchors here, invoke a method that wipes `anchors.dat`.

Reply:
If a node crash happens between wiping `anchors.dat` and creating a new one, the anchors data would be lost.